### ISSUE TITLE
TODO

### ISSUE THREAD

**Issue by Bili-Sakura** [Bili-Sakura/pytorch-image-translation-models#21]
- [x] Integrate DDIB model ckpt on hf
- [x] Integrate CDTSDE ckpt
- [x] add more metrics and categorized, and wrap as a metric class
- [x] remove src/training folder

**Comment by Bili-Sakura** [comment]
SiD2 scheduler 
```python
from diffusers import SchedulerMixin, ConfigMixin
from diffusers.utils import BaseOutput
import torch
import math

class SiD2Scheduler(SchedulerMixin, ConfigMixin):
    """
    Exact SiD2 continuous scheduler (t ∈ [0,1]).
    Drop-in replacement for DDPMScheduler in training loops.
    """
    _compatibles = ["DDPMScheduler"]
    config_name = "sid2_scheduler_config.json"

    def __init__(
        self,
        image_size: int = 512,
        num_train_timesteps: int = 1000,      # only used for discrete sampling mode
        logsnr_min: float = -10.0,
        logsnr_max: float = 10.0,
    ):
        super().__init__()
        self.register_config(
            image_size=image_size,
            num_train_timesteps=num_train_timesteps,
            logsnr_min=logsnr_min,
            logsnr_max=logsnr_max,
        )
        self.image_size = image_size

    def cosine_interpolated_logsnr(self, t: torch.Tensor) -> torch.Tensor:
        """Exact formula from SiD2 paper Appendix B"""
        log_change_high = math.log(self.image_size) - math.log(512)
        log_change_low  = math.log(self.image_size) - math.log(32)
        b = math.atan(math.exp(-0.5 * self.config.logsnr_max))
        a = math.atan(math.exp(-0.5 * self.config.logsnr_min)) - b
        logsnr_cosine = -2.0 * torch.log(torch.tan(a * t + b))
        logsnr_high = logsnr_cosine + log_change_high
        logsnr_low  = logsnr_cosine + log_change_low
        return (1 - t) * logsnr_high + t * logsnr_low

    def add_noise(
        self,
        original_samples: torch.Tensor,
        noise: torch.Tensor,
        timesteps: torch.Tensor,          # float t ∈ [0,1] or integer (auto-scaled)
    ) -> torch.Tensor:
        if timesteps.dim() == 0 or timesteps.shape[0] == 1:
            t = timesteps.to(original_samples.device).float()
        else:
            t = timesteps.to(original_samples.device).float()
        if t.max() > 1.0:                     # user passed integer timesteps
            t = t / self.config.num_train_timesteps

        logsnr = self.cosine_interpolated_logsnr(t).view(-1, 1, 1, 1)
        alpha_bar = 1.0 / (1.0 + torch.exp(-logsnr))
        sqrt_alpha = alpha_bar.sqrt()
        sqrt_one_minus_alpha = (1.0 - alpha_bar).sqrt()
        return sqrt_alpha * original_samples + sqrt_one_minus_alpha * noise

    # For completeness (sampling side)
    def get_velocity(self, sample: torch.Tensor, noise: torch.Tensor, timesteps: torch.Tensor):
        # v = α√ * ε - √(1-α) * x   (same as Diffusers v-prediction)
        logsnr = self.cosine_interpolated_logsnr(timesteps.view(-1, 1, 1, 1))
        alpha_bar = 1.0 / (1.0 + torch.exp(-logsnr))
        return alpha_bar.sqrt() * noise - (1.0 - alpha_bar).sqrt() * sample

    # You can add .step() for sampling later if needed
```
